app/src/main/java/uk/gov/onelogin/appcheck/FirebaseAppCheck.kt

```diff
@@ -2,31 +2,34 @@ package uk.gov.onelogin.appcheck
 
 import android.content.Context
 import com.google.firebase.Firebase
+import com.google.firebase.FirebaseException
 import com.google.firebase.appcheck.AppCheckProviderFactory
 import com.google.firebase.appcheck.appCheck
 import com.google.firebase.initialize
+import javax.inject.Inject
+import kotlinx.coroutines.tasks.await
+import uk.gov.android.authentication.integrity.appcheck.AppChecker
+import uk.gov.android.authentication.integrity.model.AppCheckToken
 
-class FirebaseAppCheck : AppCheck {
+class FirebaseAppCheck @Inject constructor(
+    private val appCheckFactory: AppCheckProviderFactory
+) : AppChecker {
     override fun init(
-        context: Context,
-        appCheckFactory: AppCheckProviderFactory
+        context: Context
     ) {
         Firebase.initialize(context)
         Firebase.appCheck.installAppCheckProviderFactory(
             appCheckFactory
         )
     }
 
-    override fun getAppCheckToken(
-        onSuccess: (String) -> Unit,
-        onFailure: (Exception) -> Unit
-    ) {
-        Firebase.appCheck.getAppCheckToken(false)
-            .addOnSuccessListener { token ->
-                onSuccess(token.token)
-            }
-            .addOnFailureListener { e ->
-                onFailure(e)
-            }
+    override suspend fun getAppCheckToken(): Result<AppCheckToken> {
+        return try {
+            Result.success(
+                AppCheckToken(Firebase.appCheck.getAppCheckToken(false).await().token)
```

Comment:
It might be worth pulling out the Firebase.appCheck to a class property to simplify this call chain a bit.